testing/trino-product-tests/src/main/java/io/trino/tests/jdbc/TestExternalAuthorizerOAuth2.java

```diff
@@ -87,6 +87,28 @@ public void shouldAuthenticateAndExecuteQuery()
         }
     }
 
+    @Test(groups = {OAUTH2, PROFILE_SPECIFIC_TESTS})
```

Comment:
Does this test exposes a bug or is it a test coverage? If the latter please merge this commit with the commit that fixes the issue.

Reply:
It's both as always. It exposes the bug and it adds a test coverage on product test level (e2e). Still I think that your idea of adding this test in a separate commit makes a lot of sense, as it shows that this test has caught a real case at least once.

Reply:
So before the fix, the would fail. In order to properly expose the failure it should show how it fails before the fix. So for example a line like this:
```
assertThatNoException().isThrownBy(() -> checkLastLoginActionExecutionFailures());
```
would actually expect failure instead.